at_client/src/main/java/org/atsign/client/util/KeysUtil.java

```diff
@@ -18,68 +18,93 @@
 public class KeysUtil {
     private static final ObjectMapper mapper = new ObjectMapper();
 
-    private static final String rootFolder = System.getProperty("user.dir") + "/keys/";
+    private static String rootFolder = System.getProperty("user.home") + "/.atsign/keys/";
 
-    public  static final String pkamPublicKeyName = "aesPkamPublicKey";
-    public  static final String pkamPrivateKeyName = "aesPkamPrivateKey";
-    public  static final String encryptionPublicKeyName = "aesEncryptPublicKey";
-    public  static final String encryptionPrivateKeyName = "aesEncryptPrivateKey";
-    public  static final String selfEncryptionKeyName = "selfEncryptionKey";
+    public static final String pkamPublicKeyName = "aesPkamPublicKey";
+    public static final String pkamPrivateKeyName = "aesPkamPrivateKey";
+    public static final String encryptionPublicKeyName = "aesEncryptPublicKey";
+    public static final String encryptionPrivateKeyName = "aesEncryptPrivateKey";
+    public static final String selfEncryptionKeyName = "selfEncryptionKey";
 
     public static void saveKeys(AtSign atSign, Map<String, String> keys) throws Exception {
-        _makeRootFolder();
-        File file = getKeysFile(atSign);
-        System.out.println("Saving keys to " + file.getAbsolutePath());
-        if (file.exists()) {
-            //noinspection ResultOfMethodCallIgnored
+        rootFolder += atSign + "_key.atKeys";
```

Comment:
Every time saveKeys runs you are modifying rootFolder. There was a reason it was declared final :-)

Suggest instead you make a new variable here called 'keysFile' i.e.
`String keysFileName = rootFolder + atSign + keysFileSuffix;`